app/README.md

```diff
@@ -0,0 +1,105 @@
+# App modules
+
+App modules are subsections of the main site, dealing with a specific area , e.g locations or moves. They are constructed using Express routers.
+
+They can have a variety of different parts, in either a root file, or a folder with multiple files, and exposed via an `index.js` in the folder.
+
+- controllers
+  - Express
+- constants
+  - Predefined default settings for a module, including routing paths, actions, and other defaults
+- fields
+  - Form-wizard field definitions
+- middleware
+  - Express middleware for the module, rendering against the views
+- steps
+  - Form-wizard step definitions
+- validators
+  - Form-wizard field validator functions
+- views
+  - Nunjucks templates
+- index (routing)
+  - Main entrypoint for the module, tying together all the parts of the module in the form of an Express router
+
+## Index / Routing Setup
+
+### params middleware
+
+- set req vars
+  - **setDateRange**
+    - req.period
+    - req.dateRange
+- redirect
+
+### shared middleware
+
+- **saveUrl**
+
+### routing middleware
+
+#### types
+
+- redirects
+- simple routes
+- routes with form-wizard
+- sub-routers
+- arrays of grouped middleware
+  - collections middleware (`setActions`,`setContext`,`setPagination`)
+  - data setup, retrieval and transformation middleware (`setBody` `setFilter` `setResults`)
+
+#### router.`HTTP_METHOD`
+
+##### Generic middleware
+
+- **path/with/:tokens**
+- **protectRoute**
+- redirects
+
+##### Collections setup
+
+- setActions
+- setContext
+  - permission
+  - text
+  - href
+- setPagination
+
+##### Data setup, retrieval and transformation
+
+- setBody*Something*
+  - sets `req.body.something`
+    - structure includes `status`, `sortBy`, `sortDirection`, `[locationType]`
+- set*Something*Filters
+  - calls services
+  - uses body prop
+  - sets `req.filter`
+- set*Something*Results
+  - calls services
+  - uses body prop
+  - sets `req.resultsSomething` or `req.resultsSomethingAsTable`
+
+##### Render
+- `render` method from controller, e.g. `home/controllers.js:dashboard`
+  - `res.render` with template e.g. `home/dashboard`
+    - actions
```

Comment:
Might be worth being more generic in this case and referring to the [express render docs](https://expressjs.com/en/api.html#res.render) and just document that this is called with any locals required for that view.